examples/sparse/hetero-rgcn.py

```diff
@@ -0,0 +1,320 @@
+        for ntype, num_nodes in ntype_num.items():
+            embed = nn.Parameter(th.Tensor(num_nodes, self.embed_size))
+            nn.init.xavier_uniform_(embed, gain=nn.init.calculate_gain("relu"))
+            self.embeds[ntype] = embed
+
+    def forward(self):
+        return self.embeds
+
+
+class HeteroRelationalGraphConv(nn.Module):
+    r"""HeteroRelational graph convolution layer.
+
+    Parameters
+    ----------
+    in_size : int
+        Input feature size.
+    out_size : int
+        Output feature size.
+    relation_names : list[str]
+        Relation names.
+    """
+
+    def __init__(
+        self,
+        in_size,
+        out_size,
+        relation_names,
+    ):
+        super(HeteroRelationalGraphConv, self).__init__()
+        self.in_size = in_size
+        self.out_size = out_size
+        self.relation_names = relation_names
+
+        ########################################################################
+        # (HIGHLIGHT) HeteroGraphConv is a graph convolution operator over
+        # heterogeneous graphs. A dictionary is passed where the key is the
+        # relation name and the value is the insatnce of conv layer.
+        ########################################################################
+        self.W = nn.ModuleDict(
+            {str(rel): nn.Linear(in_size, out_size) for rel in relation_names}
+        )
+
+        self.dropout = nn.Dropout(0.0)
+
+    def forward(self, A, inputs):
+        """Forward computation
+
+        Parameters
+        ----------
+        A : Hetero Sparse Matrix
+            Input graph.
+        inputs : dict[str, torch.Tensor]
+            Node feature for each node type.
+
+        Returns
+        -------
+        dict[str, torch.Tensor]
+            New node features for each node type.
+        """
+        hs = {}
+        for rel in A:
+            src_type, edge_type, dst_type = rel
+            if dst_type not in hs:
+                hs[dst_type] = th.zeros(
+                    inputs[dst_type].shape[0], self.out_size
+                )
+            ####################################################################
+            # (HIGHLIGHT) Sparse library use hetero sparse matrix to present
+            # heterogeneous graphs. A dictionary is passed where the key is
+            # the tuple of (source node type, edge type, destination node type)
+            # and the value is the sparse matrix contructed from the key on
+            # global graph. The convolution operation is the multiplication of
+            # sparse matrix and convolutional layer.
+            ####################################################################
+            hs[dst_type] = hs[dst_type] + (
+                A[rel].T @ self.W[str(edge_type)](inputs[src_type])
+            )
+            hs[dst_type] = F.relu(hs[dst_type])
+
+        return hs
+
+
+class EntityClassify(nn.Module):
+    def __init__(
+        self,
+        in_size,
+        out_dim,
```

Comment:
nit: maybe just out_size?

Reply:
I follow the API in https://github.com/dmlc/dgl/blob/master/examples/core/rgcn/hetero_rgcn.py#L285, should I fix them  all ?

Reply:
Yes, please, but fix that one in a new PR.